### PR TITLE
cluster analysis implementation

### PR DESCRIPTION
**Pull request by cansavvy**
I haven't added the html output for these yet, but will run them and add them later. Besides the two Rmds for each dataset, this PR also includes moving `clustering_statistics_functions.R` to `scripts/util` because when I merged it with the master it didn't go to the right place.

### CODE REVIEW

File: arnon_clustering_statistics.Rmd [hunk 1]
```diff
@@ -0,0 +1,219 @@
+---
+title: "Arnon Normalization Evaluation"
+author: "C. Savonen CCDL for ALSF"
+date: "12/17/2018"
+output: html_document
+---
+### Purpose: compare normalization methods for Arnon et al dataset
+
```
Comment (jaclyn-taroni) [inline arnon_clustering_statistics.Rmd:8]: If you're going to go the notebook route, you might as well put the full citation for this paper here.

File: arnon_clustering_statistics.Rmd [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+# Read in each of the normalization files
+normalized.data <- lapply(normalized.files, function(file) {
+                          readr::read_tsv(file.path("arnon_data",
+                                                    "normalized_arnon",
+                                                    file))
+                          })
+
+# Keep all the gene lists
+genes <- lapply(normalized.data, function(x) x[,1])
```
Comment (jaclyn-taroni) [inline arnon_clustering_statistics.Rmd]:
```suggestion
genes <- lapply(normalized.data, function(x) x[, 1])
```

File: arnon_clustering_statistics.Rmd [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+# Plate batch info as a factor
+plate.batch <- as.factor(meta$cohort.ch1)
+```
+
+Run tsne clustering for each dataset:
+```{r Make clusters with tsne}
+# Run tsne on each dataset and extract
+tsne <- lapply(normalized.data, function(dataset) {
+               # Run tsne
+               tsne.res <- Rtsne::Rtsne(t(dataset[,-1]), check_duplicates = FALSE)
```
Comment (jaclyn-taroni) [inline arnon_clustering_statistics.Rmd:74]:
```suggestion
               tsne.res <- Rtsne::Rtsne(t(dataset[, -1]), check_duplicates = FALSE)
```

I assume this means that `Rtsne` expects features to be columns, is this correct?

File: arnon_clustering_statistics.Rmd [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+# Extract the legend
+legend <- cowplot::get_legend(cell.type.plots[[1]])
+
+# Surpressing the legend in the files
+batch.plots <- lapply(batch.plots, function(a.plot) {
+                      a.plot + theme(legend.position = 'none')
+                      })
+
+# Put all plots and legend together
+main.plot <- cowplot::plot_grid(
```
Comment (jaclyn-taroni) [inline arnon_clustering_statistics.Rmd:139]: Some spacing issues throughout this

File: arnon_clustering_statistics.Rmd [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+                          DimPlot(dataset, cell.types, xlabel = "tsne dim 1",
+                                  ylabel = "tsne dim 2", name = set.name)
+                    })
+
+# Extract the legend
+legend <- cowplot::get_legend(cell.type.plots[[1]])
+
+# Surpressing the legend in the files
+cell.type.plots <- lapply(cell.type.plots, function(a.plot) {
+                          a.plot + theme(legend.position = 'none')
```
Comment (jaclyn-taroni) [inline arnon_clustering_statistics.Rmd:101]: Can you move this bit to above (lines 92-93)

```R
DimPlot(dataset, cell.types, xlabel = "tsne dim 1",  
        ylabel = "tsne dim 2", name = set.name) +
  theme(legend.position = 'none')
```

And snag the legend some other way?

File: arnon_clustering_statistics.Rmd [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+                          
+                        # Return data frame of combined results  
+                        data.frame(knn.results, kmeans.results)
+                        })
+```
+
+Plot all the cell type statistics on a boxplot
+```{r Print the cell type plots}
+# Melt the  list into one data.frame
+cell.type.df <- reshape::melt(cell.type.results)
```
Comment (jaclyn-taroni) [inline arnon_clustering_statistics.Rmd:182]: `reshape` or `reshape2`?

File: arnon_clustering_statistics.Rmd [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+                          DimPlot(dataset, plate.batch, xlabel = "tsne dim 1",
+                                  ylabel = "tsne dim 2", name = set.name)
+                    })
+
+# Extract the legend
+legend <- cowplot::get_legend(cell.type.plots[[1]])
+
+# Surpressing the legend in the files
+batch.plots <- lapply(batch.plots, function(a.plot) {
+                      a.plot + theme(legend.position = 'none')
```
Comment (jaclyn-taroni) [inline arnon_clustering_statistics.Rmd:135]: Same comment about suppressing legends as above

**Comment by jaclyn-taroni** [comment]
Documenting our in-person discussion here. Two points came up:

- Given the level of repetition in these notebooks, it would probably be better to make this into a script that accepts the directories, etc. as arguments
- The predictions are using the t-SNE components as features. Since the idea is to test the normalization methods, rather than a dimensionality reduction technique, it is a good idea to try clustering and prediction using the gene expression matrix (some filtering likely required).

**Comment by jaclyn-taroni** [comment]
I revisited [Kiselev, et al.](https://doi.org/10.1038/s41576-018-0088-9) after we chatted and it seems that dimensionality reduction is standard practice prior to clustering (this makes sense). I'm still not sure about t-SNE over PCA though.

**Comment by cansavvy** [comment]
^Makes sense I suppose. What about umap?

**Comment by jaclyn-taroni** [comment]
If we're going to pick one (and I think that's wise), it makes sense to me to use PCA.

**Comment by cansavvy** [comment]
Because of the large shift in implementation of these analyses and that [incorporation of cluster analyses in pipeline](https://github.com/cansav09/scRNA-seq_workflow/pull/15) covers this same material, I'm going to go ahead and close this issue.

**Pull request closed without merging**